core/os/os.cpp

```diff
@@ -391,6 +391,42 @@ void OS::set_has_server_feature_callback(HasServerFeatureCallback p_callback) {
 	has_server_feature_callback = p_callback;
 }
 
+String OS::get_architecture_name() const {
```

Comment:
Would be good if `has_feature` could be refactored to use this.

Reply:
This occurred to me too, I wrote a comment but then deleted it. We'd still need `has_feature` to have more than just this method, for example if we still want to have `x86` return true on x86_32 and x86_64.

Reply:
I still think this should be done. Maintaining all these defs in two places with a slightly different set of conditions doesn't seem good to me.

Maybe it's also time to review what `OS::has_feature` does, I'm not sure it's useful to have e.g. `x86`. It's not that hard to check both `x86_64` and `x86_32` if that's what one needs to know (and it's not useful e.g. for GDExtension libraries as they won't be compatible for these two `x86` arches).

Reply:
I also agree that we probably don't need to have generic `x86` and `arm` feature tags. If you really need a project setting to apply to both 32-bit and 64-bit x86/ARM, you could overrid it twice.